src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -873,6 +923,92 @@ NodeIsFieldStore(Node *node)
 }
 
 
+/*
+ * MergeQuerySupported does check for a MERGE command in the query, if it finds
+ * one, it will verify the below criteria
+ * - Supported tables and combinations in ErrorIfMergeHasUnsupportedTables
+ * - Distributed tables requirements in ErrorIfDistTablesNotColocated
+ * - Checks target-lists and functions-in-quals in TargetlistAndFunctionsSupported
+ */
+static DeferredErrorMessage *
+MergeQuerySupported(Query *originalQuery,
```

Comment:
again, could go to `planner/merge.c` or such

Reply:
Done.